### PR TITLE
[Documentation]: Added description for NNCF PTQ

### PR DESCRIPTION
**Pull request by AlexKoff88**
### Details:
* Create documentation for NNCF PTQ feature
* Revised model optimization related documents

### CODE REVIEW

File: docs/optimization_guide/nncf/ptq/code/ptq_openvino.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#! [dataset]
+import nncf
+import torch
+
+calibration_loader = torch.utils.data.DataLoader(...)
+
+def transform_fn(data_item):
+    images, _ = data_item
+    return images.numpy()
```
Comment (alexsu52) [inline docs/optimization_guide/nncf/ptq/code/ptq_openvino.py:12]: return images

File: docs/optimization_guide/nncf/ptq/code/ptq_tensorflow.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+#! [dataset]
+import nncf
+import tensorflow_datasets as tfds
+
+calibration_loader = val_dataset = tfds.load(...)
```
Comment (alexsu52) [inline docs/optimization_guide/nncf/ptq/code/ptq_tensorflow.py:8]:
```suggestion
calibration_loader = tfds.load(...)
```

File: docs/optimization_guide/nncf/ptq/ptq_introduction.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   basic_qauntization_flow
+   quantization_w_accuracy_control
+
+@endsphinxdirective
+
+Neural Network Compression Framework (NNCF) provides a new post-training quantization API available in Python that is aimed at reusing the code for model training or validation that usually available with the model in the source framework, for example, PyTorch* or TensroFlow*. The API is cross-framework and currently supports models representing in the following frameworks: PyTorch, TensorFlow 2.x, ONNX, and OpenVINO.
```
Comment (alexsu52) [inline docs/optimization_guide/nncf/ptq/ptq_introduction.md:14]:
```suggestion
Neural Network Compression Framework (NNCF) provides a new post-training quantization API available in Python that is aimed at reusing the code for model training or validation that is usually available with the model in the source framework, for example, PyTorch* or TensroFlow*. The API is cross-framework and currently supports models representing in the following frameworks: PyTorch, TensorFlow 2.x, ONNX, and OpenVINO. 
```

File: docs/optimization_guide/nncf/ptq/quantization_w_accuracy_control.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Quantizing with accuracy control {#quantization_w_accuracy_control}
+
+## Introduction
+
+This is the advanced quantization flow that allows to apply 8-bit quantization to the model with control of accuracy metric. This is achieved by keeping the most impactful operations within the model in the original precision. The flow is based on the [Basic 8-bit quantization](@ref basic_qauntization_flow) and has the following differences:
+* Besided the calibration dataset, a **validation dataset** is required to compute accuracy metric. They can refer to the same data in the simplest case.
+* **Validation function**, used to compute accuracy metric is required. It can be a function that is already available in the source framework or a custom function.
+* Since accuracy validation is run several times during the quantization process, quantization with accuracy control can take more time than the [Basic 8-bit quantization](@ref basic_qauntization_flow) flow.
+* The resulted model can provide smaller performance improvement than the [Basic 8-bit quantization](@ref basic_qauntization_flow) flow bacause some of the operations are kept in the original precision.
```
Comment (alexsu52) [inline docs/optimization_guide/nncf/ptq/quantization_w_accuracy_control.md:9]:
```suggestion
* The resulted model can provide smaller performance improvement than the [Basic 8-bit quantization](@ref basic_qauntization_flow) flow because some of the operations are kept in the original precision.
```

File: docs/optimization_guide/nncf/ptq/code/ptq_openvino.py (hunk 1 above)
Comment (AlexKoff88) [reply docs/optimization_guide/nncf/ptq/code/ptq_openvino.py:12]: Numpy is the source format for OpenVINO. Even so, torch.Tensor is acceptable, I would like to highlight it.

**Review by alexsu52 — APPROVED** [review]

File: docs/optimization_guide/nncf/ptq/basic_quantization_flow.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+python3 -m venv nncf_ptq_env
+```
+You should also install all the packages required to instantiate the model object, for example DL fraework. After that you need to install NNCF on top of the environment:
+```bash
+pip install nncf
+```
+
+## Prepare calibration dataset
+
+At this step, you should create an instance of `nncf.Dataset` class that represents the calibration dataset. The `nncf.Dataset` class can be a wrapper over the framework dataset object that is used for model training or validation. The class constructor receives dataset object and the so-called transformation function. For example, if you use PyTorch, you can pass an instance of `torch.utils.data.DataLoader` object.
```
Comment (tsavina) [inline docs/optimization_guide/nncf/ptq/basic_quantization_flow.md:23]:
```suggestion
At this step, create an instance of the `nncf.Dataset` class that represents the calibration dataset. The `nncf.Dataset` class can be a wrapper over the framework dataset object that is used for model training or validation. The class constructor receives the dataset object and the transformation function. For example, if you use PyTorch, you can pass an instance of the `torch.utils.data.DataLoader` object. 
```

**Review by kblaszczak-intel — APPROVED** [review]
With tsavina's comments, I think we can merge it. Later improvements can be made when the bulk of changes is already in place.

File: docs/optimization_guide/ptq_introduction.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+.. toctree::
+   :maxdepth: 1
+   :hidden:
+
+   pot_introduction
+   nncf_ptq_introduction
+
+@endsphinxdirective
+
+Post-training model optimization is the process of applying special methods that transform the model into a more hardware-friendly representation without retraining or fine-tuning. The most popular and widely-spread method here is 8-bit post-training quantization because it is:
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:14]:
```suggestion
Post-training model optimization is the process of applying special methods that transform the model into a more hardware-friendly representation without retraining or fine-tuning. The most popular and widely-spread method is 8-bit post-training quantization as it has several important advantages:
```

File: docs/optimization_guide/ptq_introduction.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* Easy-to-use
+* It does not hurt accuracy a lot
+* Provides significant performance improvement
+* Suites to many HW available in stock since most of it supports 8-bit computation natively
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:18]:
```suggestion
* It is easy-to-use.
* It does not hurt accuracy a lot.
* It provides significant performance improvement.
* It suites many hardware available in stock since most of them support 8-bit computation natively.
```

File: docs/optimization_guide/ptq_introduction.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+8-bit integer quantization lowers precision of weights and activations to 8 bits which leads to almost 4x reduction in the model footprint and significant improvements in inference speed mostly due to lower throughput required for the inference. This lowering step is done offline, i.e. before the actual inference, so that the model gets transformed to the quantized representation. The process does not require a training dataset or a training pipeline in the source DL framework.
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:20]:
```suggestion
8-bit integer quantization lowers the precision of weights and activations to 8 bits, which leads to almost 4x reduction in the model footprint and significant improvements in inference speed, mostly due to lower throughput required for the inference. This lowering step is done offline, before the actual inference, so that the model gets transformed into the quantized representation. The process does not require a training dataset or a training pipeline in the source DL framework. 
```

File: docs/optimization_guide/ptq_introduction.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+* Provides significant performance improvement
+* Suites to many HW available in stock since most of it supports 8-bit computation natively
+
+8-bit integer quantization lowers precision of weights and activations to 8 bits which leads to almost 4x reduction in the model footprint and significant improvements in inference speed mostly due to lower throughput required for the inference. This lowering step is done offline, i.e. before the actual inference, so that the model gets transformed to the quantized representation. The process does not require a training dataset or a training pipeline in the source DL framework. 
+
+![](../img/quantization_picture.png)
+
+To apply post-training methods in OpenVINO&trade;, you need:
+* A floating-point precision model, FP32 or FP16, converted into the OpenVINO&trade; Intermediate Representation (IR) format that can be run on CPU.
+* A representative calibration dataset representing a use case scenario, for example, 300 samples.
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:26]:
```suggestion
* A representative calibration dataset, representing a use case scenario, for example, of 300 samples.
```

File: docs/optimization_guide/ptq_introduction.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+* In case of accuracy constraints, a validation dataset and accuracy metrics should be available.
+
+Currently, there are OpenVINO&trade; Toolkit provides two workflows that provide post-training quantization capabilities:
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:29]:
```suggestion
Currently, OpenVINO provides two workflows with post-training quantization capabilities:
```

File: docs/optimization_guide/ptq_introduction.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+* [Post-training Quantization w/ POT](@ref pot_introduction) - works with models in OpenVINO Intermediate Representation (IR) only.
+* [Post-training Quantization w/ NNCF](@ref nncf_ptq_introduction) - cross-framework solution for model optimization that provides a new simple API for post-training quantization.
```
Comment (tsavina) [inline docs/optimization_guide/ptq_introduction.md:31]:
```suggestion
* [Post-training Quantization with POT](@ref pot_introduction) - works with models in OpenVINO Intermediate Representation (IR) only.
* [Post-training Quantization with NNCF](@ref nncf_ptq_introduction) - cross-framework solution for model optimization that provides a new simple API for post-training quantization.
```

File: docs/optimization_guide/nncf/ptq/basic_quantization_flow.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
+# Basic quantization flow {#basic_qauntization_flow}
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply docs/optimization_guide/nncf/ptq/basic_quantization_flow.md:1]: that's a feature)

**Comment by AlexKoff88** [comment]
Applied most of the comments.

**Review by tsavina — APPROVED** [review]
Let's merge and fix the rest in separate PRs
